### DISCUSSION TITLE
JupyterHub support. SEB for MacOS 3.0 browser problem with PDFs

### DISCUSSION THREAD

**Discussion by rafael-palacios** [SafeExamBrowser/seb-mac#156] [General]
JupyterHub is a freeware web-based environment for Programming.
It supports Python, Java, Julia, etc.... It is becoming the standard for teaching programming and specially for exams.
SEB works perfectly for this because teachers can prepare a config file that takes students directly to the JupyterHub server where they find their exams and work on the server directly.
SEB for windows 3.3.1 works perfectly
SEB for macOS 2.3.2 works perfectly. 
However, SEB for MacOS 3.0 uses a new embedded browser that doesn't support PDF files.

**Comment by danschlet** [comment]
Of course the modern WebKit browser engine still supports rendering PDFs. I guess there’s some incompatibility with how displaying PDFs is implemented in JupyterHub. Did you try to change your SEB settings to use the classic browser engine? See Preferences/Browser/"Select Browser Engine Policy"/Force Classic?
For further investigating this incompatibility, we would need some testing access to a JupyterHub instance and detailed steps to reproduce the issue.

**Reply by rafael-palacios** [reply]
You are right!!
Selecting "Classic Browser" option SEB 3.0 for MacOS works with JupyterHub PDF files.
Thank you for your help.
